meson.build

```diff
@@ -3,13 +3,38 @@ project('abseil-cpp', 'cpp',
   license : 'Apache-2.0',
   default_options : [
     'cpp_std=c++11',
+    'warning_level=0'
   ])
 
 cpp = meson.get_compiler('cpp')
 
 cpp_flags = [
-  '-Wno-sign-compare'
+  '-Wcast-qual',
+  '-Wconversion-null',
+  '-Wformat',
+  '-Wformat-security',
+  '-Wmissing-declarations',
+  '-Woverlength-strings',
+  '-Wpointer-arith',
+  '-Wundef',
+  '-Wunused-local-typedefs',
+  '-Wunused-result',
+  '-Wvarargs',
+  '-Wvla',
+  '-Wwrite-strings',
+  '-DNOMINMAX',
 ]
+
+if build_machine.cpu_family() in ['x86_64', 'amd64', 'AMD64']
+  cpp_flags += [ '-maes', '-msse4.1' ]
+elif build_machine.cpu_family().to_lower().contains('arm') or build_machine.cpu_family() == 'aarch64'
+  if cpp.sizeof('void*') == 8
+    cpp_flags += [ '-march=armv8-a+crypto' ]
+  elif cpp.sizeof('void*') == 4
+    cpp_flags += [ '-mfpu=neon' ]
```

Comment:
Does this compile on ARMv7?

Reply:
The gcc on Raspberry Pi at last accepts it without problems. It's probably ARMv6 that would be problematic. If someone runs into this then they can submit patches.

Reply:
Unfortunately I don't have an ARMv7 machine handy now to check, just `x86_64` and `aarch64`

Reply:
Doesn't it require `-funsafe-math-optimizations` ? IIRC that was the case a while ago.

Reply:
I'm not sure - I could just support the two cpu families which I know it compiles against and omit the `arm` completely.

Reply:
I would leave it, but fix the condition to make sense (ie do not look for an aarch64 cpu with 4 bytes pointers).